src/TypeScriptInternal.ts

```diff
@@ -66,3 +66,89 @@ export function getTextOfJsxAttributeName(node: ts.JsxAttributeName): string {
     ? ts.idText(node)
     : `${ts.idText(node.namespace)}:${ts.idText(node.name)}`
 }
+
+type ObjectLiteralElementWithName = ts.ObjectLiteralElement & {
+  name: ts.PropertyName
+  parent: ts.ObjectLiteralExpression | ts.JsxAttributes
+}
+export function getContainingObjectLiteralElement(
```

Comment:
Link to original source in comment? I probably didn't do this before but it might be a good idea

Reply:
Probably a good idea. I'll look them up and add the links.